sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/catalog/SessionCatalog.scala

```diff
@@ -705,7 +705,7 @@ class SessionCatalog(
             schema = plan.output.toStructType)
       }.getOrElse(throw new NoSuchTableException(globalTempViewManager.database, table))
     } else {
-      getTableMetadata(name)
+      getTableRawMetadata(name)
```

Comment:
why do we need this change?

Reply:
AlterTableAddColumnsCommand will mess up the original schema or I just get the raw schema there？

Reply:
Ideally, this method should return the raw type of a table

Reply:
It is also related to the API `spark.catalog.listColumns` which returns a list of columns for the given table/view

Reply:
It's a bit weird that `getTempViewOrPermanentTableMetadata` and `getTableMetadata` is not consistent. How many caller sides we need to update in this case?

Reply:
I have checked all the callers,` spark.catalog.listColumns` and AlterTableAddColumnsCommand only will directly reference the schema here.

Reply:
if it's only 2 places, let's update the caller sides.

Reply:
Ok.
Just a little wired that just change the data type back and forth in the caller